lib/3scale/backend/worker_async.rb

```diff
@@ -94,23 +98,14 @@ def clear_queue
         barrier.wait
       end
 
-      def start_thread_to_fetch_jobs
-        Thread.new do
-          Sync { @job_fetcher.start(@jobs) }
-        end
+      def start_to_fetch_jobs
+        Async(transient: true) { @job_fetcher.start(@jobs) }
```

Comment:
Why is this task transient?...

Reply:
This means when all other tasks in the reactor have finished, don't wait for this task, just kill it.

https://socketry.github.io/async/guides/asynchronous-tasks/index#reactor-lifecycle

Reply:
Yeah, that's what I read in the docs, and that makes me wonder... Don't we need the job fetching to be running forever? (even if other non-transient tasks are completed)

Maybe I just don't understand how it works, or I don't know what other jobs are running in the same reactor.

If you do understand it and it looks good, then OK :)

Reply:
there is one task fetching (this one) and another processing jobs. The processing loop also spawns individual tasks but lets leave that alone. If we don't have processing loop, we don't want to fetch jobs from redis as we will just lose them without anything to perform them.